include/deal.II/matrix_free/cuda_tensor_product_kernels.h

```diff
@@ -171,12 +185,11 @@ namespace CUDAWrappers
                            dim,
                            fe_degree,
                            n_q_points_1d,
-                           Number>::gradients(const Number *in,
+                           Number>::gradients(Number        shape_gradients[],
+                                              const Number *in,
                                               Number *      out) const
     {
-      apply<direction, dof_to_quad, add, in_place>(global_shape_gradients,
-                                                   in,
-                                                   out);
+      apply<direction, dof_to_quad, add, in_place>(shape_gradients, in, out);
```

Comment:
I know that currently it does not make any sense to distinguish between  `apply`, `values` and `gradients`. However, this is just temporal. Tomorrow, I plan to refactor the whole `cuda_tensor_product_kernels.h`-file, i.e., to split up `EvaluatorTensorProduct` into two parts - as it is the case in `MatrixFree` (actual): in that configuration references to the shape- and the gradient-values will be stored in class fields. The modifications will us allow to use collocation methods and odd-even evaluations. Because of this refactoring, I somewhat hope that the PRs will be merged today ;)